### PR TITLE
Configure Keycloak clients using keycloak_client module

### PR DESCRIPTION
**Pull request by midmarch**
- [ ] Configure `ANET-Realm`, requires [Ansible realm support](https://github.com/ansible-collections/community.general/pull/859), created issue: #15
- [x] Configure Keycloak clients, in line with [Setting up Keycloak (user authentication)](https://github.com/NCI-Agency/anet/blob/keycloak-integration/docs/keycloak.md):

  - Creates the server client: "ANET-Client"
  - Creates the public client: "ANET-Client-public"
- Introduces a dependency on [keycloak_client](https://docs.ansible.com/ansible/latest/collections/community/general/keycloak_client_module.html) which is a plugin part of the [community.general collection](https://docs.ansible.com/ansible/latest/collections/community/general/).
- [x] Move the internal Keycloak HTTP ports configuration to the Ansible Keycloak role

### CODE REVIEW

File: roles/keycloak/tasks/client-config.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    auth_password: '{{ keycloak_password }}'
+    state: present
+    # ANET-Client
+    client_id: ANET-Client-public
+    implicit_flow_enabled: no
+    standard_flow_enabled: yes
+    protocol: openid-connect
+    public_client: yes # Access Type = confidential
+    root_url: http://localhost:8080
+    service_accounts_enabled: yes
```
Comment (gjvoosten) [inline roles/keycloak/tasks/client-config.yml:51]: Should be `no` for a public client.

File: roles/keycloak/tasks/client-config.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    auth_realm: master
+    auth_username: '{{ keycloak_user }}'
+    auth_password: '{{ keycloak_password }}'
+    state: present
+    # ANET-Client
+    client_id: ANET-Client
+    implicit_flow_enabled: no
+    standard_flow_enabled: yes
+    protocol: openid-connect
+    public_client: no # Access Type = confidential
```
Comment (gjvoosten) [inline roles/keycloak/tasks/client-config.yml:18]: Add missing line:
```
    service_accounts_enabled: yes
```

File: roles/keycloak/tasks/start.yml [hunk 3]
```diff
@@ -3,3 +3,14 @@
...
+  register: keycloak_started
+
+- name: Wait for Keycloak HTTP port to listen...
+  wait_for:
+    host: localhost
+    port: '{{ keycloak_http_port }}'
+    delay: 10
+    timeout: 30
+    state: started
+    msg: 'KeyCloak HTTP port {{ keycloak_http_port }} is not listening'
```
Comment (gjvoosten) [inline roles/keycloak/tasks/start.yml:15]: `KeyCloak` ⇒ `Keycloak`

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: roles/keycloak/tasks/client-config.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+- set_fact:
+    anet_keycloak_key_server: "{{ lookup('password', '/dev/null length=32 chars=ascii_letters') }}"
+
+- name: 'Keycloak client: ANET confidential client (used by the server-side)'
```
[the remark this answers is not in the document]
Comment (midmarch) [reply roles/keycloak/tasks/client-config.yml:4]: It is required for the semicolon.

File: roles/keycloak/tasks/start.yml (hunk 3 above)
Comment (midmarch) [reply roles/keycloak/tasks/start.yml]: :heavy_check_mark:

File: roles/keycloak/tasks/client-config.yml (hunk 2 above)
Comment (midmarch) [reply roles/keycloak/tasks/client-config.yml]: :heavy_check_mark:

File: roles/keycloak/tasks/client-config.yml (hunk 1 above)
Comment (midmarch) [reply roles/keycloak/tasks/client-config.yml]: :heavy_check_mark:

**Pull request merged**